mq/main/bridge/bridge-jms/src/main/java/com/sun/messaging/bridge/service/jms/DMQ.java

```diff
@@ -535,71 +536,71 @@ public synchronized void sendMessage(Message m, String mid, DMQReason reason, Th
 
     public static void logMessage(Message msg, String mid, Link l, Logger logger) {
 
-        StringBuffer buf = new StringBuffer();
+        StringBuilder buf = new StringBuilder();
         try {
 
-            buf.append("Logging message going to DMQ for " + l);
+            buf.append("Logging message going to DMQ for ").append(l);
             buf.append("\n");
             buf.append("\tJMS Headers:");
             buf.append("\n");
             try {
-                buf.append("\tJMSMessageID=" + msg.getJMSMessageID());
+                buf.append("\tJMSMessageID=").append(msg.getJMSMessageID());
```

Comment:
It changes behavior. On exception
- before: `buf` finally is logged as
```
Unable to get JMSMessageID header from...
```
- after:
```
JMSMessageID=	Unable to get JMSMessageID
```

Reply:
@Cousjava Did you see @pzygielo's comments about changed behavior, any chance to resolve that?